dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1401,99 +1386,107 @@ protected void updateUserReferences(User userToReplace, String replacementUserId
+        final NotificationAPI notAPI = APILocator.getNotificationAPI();
+        final Connection conn = DbConnectionFactory.getConnection();
+        try {
+            final ESContentletIndexAPI indexAPI = new ESContentletIndexAPI();
+            dc.setSQL("SELECT COUNT(*) AS count FROM " + tableName);
+            final List<Map<String, String>> results = dc.loadResults();
+            final long totalCount = Long.parseLong(results.get(0).get("count"));
+            try (final PreparedStatement ps = conn.prepareStatement("SELECT inode FROM " +
+                    tableName)) {
+                List<Contentlet> contentToIndex = new ArrayList<>();
+                final int batchSize = 100;
+                int completed = 0;
+                try (final ResultSet rs = ps.executeQuery()) {
+                    for (int i = 1; rs.next(); i++) {
+                        final String inode = rs.getString("inode");
+                        cc.remove(inode);
+                        final Contentlet content = find(inode);
+                        contentToIndex.add(content);
+                        contentToIndex.addAll(indexAPI.loadDeps(content));
+                        if (i % batchSize == 0) {
+                            indexAPI.indexContentList(contentToIndex, null, false);
+                            completed += batchSize;
+                            contentToIndex = new ArrayList<>();
+                            HibernateUtil.getSession().clear();
+                            Logger.info(this, String.format("Reindexing related content after " +
+                                    "deletion of user '%s'. " + "Completed: " + completed + " out of " +
+                                    totalCount, userToReplace.getUserId() + "/" + userToReplace
+                                    .getFullName()));
+                        }
+                    }
+                    // index remaining records if any
+                    if (!contentToIndex.isEmpty()) {
+                        indexAPI.indexContentList(contentToIndex, null, false);
+                    }
+                }
+            }
+            dc.setSQL("DROP TABLE " + tableName);
+            dc.loadResult();
+            Logger.info(this, String.format("Reindexing of updated related content after " +
+                    "deleting user '%s' " + " has finished successfully.", userToReplace.getUserId() +
+                    "/" + userToReplace.getFullName()));
+        } catch (Exception e) {
+            Logger.error(this.getClass(), e.getMessage(), e);
+            notAPI.error(String.format("Unable to reindex updated related content for deleted " +
+                    "user '%s'. " + "Please run a full Reindex.", userToReplace.getUserId() + "/" +
+                    userToReplace.getFullName()), user.getUserId());
+        } finally {
+            if (null != conn) {
+                try {
+                    conn.close();
+                } catch (SQLException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)